packages/adapters/src/docker-instance-adapter.ts

```diff
@@ -169,7 +169,7 @@ IComponent {
 
     // eslint-disable-next-line complexity
     async run(config: InstanceConfig, instancesServerPort: number, instanceId: string): Promise<ExitCode> {
-        if (config.type !== "docker") {
+        if (!("container" in config)) { //config.type !== "docker"
```

Comment:
```suggestion
        if (!(config.type == "docker" && ("container" in config))) {
```

?

Reply:
I changed it because the creation of new type,  just for displaying, was connected with the problem  of possible "docker" type without container and config, which are used in this func.

Reply:
You are right, yours is more readable.